modules/bibindex/lib/bibindex_engine_utils.py

```diff
@@ -490,12 +490,7 @@ class UnknownTokenizer(Exception):
 
 def list_union(list1, list2):
     "Returns union of the two lists."
-    union_dict = {}
-    for e in list1:
-        union_dict[e] = 1
-    for e in list2:
-        union_dict[e] = 1
-    return union_dict.keys()
+    return list(set().union(list1, list2))
```

Comment:
Looks good to me: the same effect but more performant.
Even though probably I would have written: `list(set(list1).union(list2))`